vm/vm_jit.go

```diff
@@ -1,31 +1,376 @@
+// +build evmjit
+
 package vm
 
-import "math/big"
+/*
+#include <stdint.h>
+#include <stdlib.h>
+
+struct evmjit_result
+{
+	int32_t  returnCode;
+	uint64_t returnDataSize;
+	void*    returnData;
+};
+
+struct evmjit_result evmjit_run(void* _data, void* _env);
+
+#cgo LDFLAGS: -levmjit
+*/
+import "C"
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/state"
+	"math/big"
+	"unsafe"
+)
 
 type JitVm struct {
-	env    Environment
-	backup *Vm
+	env        Environment
+	me         ContextRef
+	callerAddr []byte
+	price      *big.Int
+	data       RuntimeData
+}
+
+type i256 [32]byte
+
+const (
+	Gas = iota
+	address
+	Caller
+	Origin
+	CallValue
+	CallDataSize
+	GasPrice
+	CoinBase
+	TimeStamp
+	Number
+	Difficulty
+	GasLimit
+	CodeSize
+
+	_size
+
+	ReturnDataOffset   = CallValue // Reuse 2 fields for return data reference
+	ReturnDataSize     = CallDataSize
+	SuicideDestAddress = address ///< Suicide balance destination address
+)
+
+type RuntimeData struct {
+	elems    [_size]i256
+	callData *byte
+	code     *byte
+}
+
+func hash2llvm(h []byte) i256 {
+	var m i256
+	copy(m[len(m)-len(h):], h) // right aligned copy
+	return m
+}
+
+func llvm2hash(m *i256) []byte {
+	if len(m) != 32 {
+		panic("I don't know Go!")
```

Comment:
i256 always has a length of 32 bytes. Why would you put this panic and check here?